### ISSUE TITLE
Parsing Kafka messages as Json Array

### ISSUE THREAD

**Issue by Gillani0** [ClickHouse/ClickHouse#5029]
Hi,

From the documentation, it's not very clear to me if we can use Json array to be parsed for the Kafka engine using ```JSONEachRow```. For instance, if I have data with the following format, how can I capture it in a table with ```Kafka engine```:

```json
 [{"p1":"v1","p2":"v1","timestamp":1555060245000},{"p1":"v2","p2":"v2","timestamp":1555060245000}]
```

PS: the data in the json array is quite flat and doesn't contain any complex structure. Cheers

**Comment by den-crane** [comment]
Just use select ... format JSONEachRow to see how CH expects data for input

```
select * from (
select 1 row, 2 int_sample, [1,2] int_array_sample
union all
select 2 row, 0 int_sample, [4,5] int_array_sample)
format JSONEachRow

{"row":1,"int_sample":2,"int_array_sample":[1,2]}
{"row":2,"int_sample":0,"int_array_sample":[4,5]}
```

**Comment by Gillani0** [comment]
Hi den,
It expects it to start with `{` and it's complaining when it starts with `[` for a Json array. 

Syd

**Comment by den-crane** [comment]
https://clickhouse.yandex/docs/en/interfaces/formats/ Check input format column
https://clickhouse.yandex/docs/en/interfaces/formats/#jsoneachrow

>It expects it to start with { and it's complaining when it starts with [ for a Json array.

Yes CH expects the data as a whole is not valid JSON

**Comment by Gillani0** [comment]
Thanks Denis. 
So there is no way to process Json arrays of objects in CH.

**Comment by bicubic** [comment]
Between this and #3579 there seems to be no way to pass a collection of records per kafka message at all.

**Comment by dengwanc** [comment]
seems there is no such way parse something like this
`{"arr1": [{"a": 1, "b": 2}]}`

**Comment by letterkey** [comment]
> seems there is no such way parse something like this
> `[{"a": 1, "b": 2}, {"a":1, "b":3}]`
> 
> @alexey-milovidov shoule we have some function like this
> 
> ```
> select JSONExtractArrayRaw('[{"a": 1, "b": 2}, {"a":1, "b":3}]')
> =>
> Array of Raw String
> ['{"a": 1, "b": 2}', '{"a":1, "b":3}']
> ```

this right

**Comment by blinkov** [comment]
@Gillani0, do you have any further questions?

**Comment by filimonov** [comment]
It will be fixed here #10163
